File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.java
import java.util.*;
class Solution {
        public int[] solution(int[] numbers, String direction) {
        int[] answer = new int[numbers.length];
        int temp = 0;
        
        if(direction.equals("right")){
            temp = numbers[numbers.length - 1];
            System.arraycopy(numbers, 0, answer, 1, numbers.length - 1);
            answer[0] = temp;
        }else{
            temp = numbers[0];
            answer = Arrays.copyOfRange(numbers, 1, numbers.length + 1);
            answer[answer.length - 1] = temp;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/120844. 배열 회전시키기/배열 회전시키기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

int* solution(int numbers[], size_t numbers_len, const char* direction) {
    int* answer = (int*)malloc(numbers_len * sizeof(int));
    int idx = 0;
    
    if(!strcmp(direction, "right")){
        answer[idx++] = numbers[numbers_len - 1];
        for(int i = 0; i < numbers_len - 1; i++){
            answer[idx++] = numbers[i];
        }
    }else {
        answer[numbers_len - 1] = numbers[0];
        for(int j = 1; j < numbers_len; j++){
            answer[idx++] = numbers[j];
        }
    }
    
    return answer;
}